grakn-dist/src/grakn

```diff
@@ -0,0 +1,662 @@
+update_classpath() {
+  # Define CLASSPATH, exclude slf4j as we use logback
+  for jar in "${GRAKN_HOME}"/services/lib/*.jar; do
+      if [[ $jar != *slf4j-log4j12* ]] ; then
+          CLASSPATH="$CLASSPATH":"$jar"
+      fi
+  done
+
+  # Add path containing grakn.properties and logback.xml
+  CLASSPATH="$CLASSPATH":"${GRAKN_HOME}"/conf
+  CLASSPATH="$CLASSPATH":"${GRAKN_HOME}"/services/grakn
+}
+
+# ================================================
+# storage helper functions
+# ================================================
+storage_start_storage() {
+  "${GRAKN_HOME}"/services/cassandra/cassandra -p $STORAGE_PID > /dev/null 2>&1
+  return $?
+}
+
+storage_is_storage_running() {
+  local status=
+  if [ -e $STORAGE_PID ] && ps -p `cat $STORAGE_PID` > /dev/null ; then
+    status=0 # storage running
+  else
+    status=1 # storage not running
+  fi
+  return $status
+}
+
+storage_wait_until_ready() {
+  local now_s=`date '+%s'`
+  local stop_s=$(( $now_s + $STORAGE_STARTUP_TIMEOUT_S ))
+  local status_thrift=
+
+  while [ $now_s -le $stop_s ]; do
+      echo -n "."
+      # The \r\n deletion bit is necessary for Cygwin compatibility
+      status_thrift=`"${GRAKN_HOME}"/services/cassandra/nodetool statusthrift 2>/dev/null | tr -d '\n\r'`
+      if [ $? -eq 0 -a 'running' = "$status_thrift" ]; then
+          return 0
+      fi
+      sleep $WAIT_INTERVAL_S
+      now_s=`date '+%s'`
+  done
+
+  return 1
+}
+
+storage_start_and_wait_until_ready() {
+  storage_is_storage_running
+  is_storage_running=$?
+
+  if [ $is_storage_running -eq 0 ]; then
+    echo "Storage is already running"
+    return 0 # storage is already running. no need to do anything
+  else
+    # otherwise attempt to start
+    echo -n "Starting Storage..."
+    storage_start_storage
+    local exec_storage_bin=$?
+
+    # check storage status and wait until ready
+    local status=
+    if [ $exec_storage_bin -eq 0 ] ; then
+      # exec storage bin successful. now wait until ready
+      storage_wait_until_ready
+      status=$?
+      echo "SUCCESS"
+    else
+      # exec storage bin failed
+      status=1
+      echo "FAILED!"
+    fi
+
+    return $status
+  fi
+}
+
+grakn_server_stop_storage() {
+  echo -n "Stopping Storage..."
+  storage_is_storage_running
+  local is_storage_running=$?
+  if [[ $is_storage_running -eq 0 ]]; then
+    if [[ -e $STORAGE_PID ]]; then
+      pid=`cat $STORAGE_PID`
+      kill "$pid"
+      while kill -0 "$pid" 2>/dev/null; do
+          sleep 0.5
+      done
+      echo "SUCCESS"
+      return 0
+    else
+      echo "FAILED!" # unable to find PID
+      return 1
+    fi
+  else
+    echo "NOT RUNNING"
+  fi
+}
+
+storage_clean_storage() {
+  rm -rf db/cassandra
+  mkdir -p db/cassandra/data db/cassandra/commitlog db/cassandra/saved_caches
+}
+
+# ================================================
+# queue helper functions
+# ================================================
+queue_start_queue() {
```

Comment:
I think this should just be queue_start instead for readability and less repetitiveness.
Also queue_check_running instead of queue_is_queue_running.

Reply:
@burukuru the idea is to "namespace" the functions - `storage_*`, `queue_*`, `grakn_*`. if you have an idea how to name them in a way that doesn't sound "weird" but still maintain organization i'm up for it